network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/BatteryImpl.java

```diff
@@ -101,6 +106,55 @@ public Battery setMaxP(double maxP) {
 
     }
 
+    private <E extends Extension<Battery>> E createActivePowerControlExtension() {
+        E extension = null;
+        ActivePowerControlAttributes attributes = resource.getAttributes().getActivePowerControl();
+        if (attributes != null) {
+            extension = (E) new ActivePowerControlImpl<>(getInjection(), attributes.isParticipate(), attributes.getDroop());
+        }
+        return extension;
+    }
+
+    @Override
+    public <E extends Extension<Battery>> void addExtension(Class<? super E> type, E extension) {
+        super.addExtension(type, extension);
+        if (type == ActivePowerControl.class) {
+            ActivePowerControl<Battery> activePowerControl = (ActivePowerControl) extension;
+            resource.getAttributes().setActivePowerControl(ActivePowerControlAttributes.builder()
+                    .participate(activePowerControl.isParticipate())
+                    .droop(activePowerControl.getDroop())
+                    .build());
+        }
+    }
+
+    @Override
+    public <E extends Extension<Battery>> E getExtension(Class<? super E> type) {
+        E extension = super.getExtension(type);
+        if (type == ActivePowerControl.class) {
+            extension = createActivePowerControlExtension();
```

Comment:
is it ok to create everytime ? Also the extesnion is already assigned in createActivePowerControlExtension, only if not null... why the two places ?

Reply:
I don't get the problem here. What do you mean with the 2 places ?